flink-libraries/flink-table/src/test/java/org/apache/flink/api/java/table/test/StringExpressionsITCase.java

```diff
@@ -121,6 +120,66 @@ public void testNonWorkingSubstring2() throws Exception {
 		resultSet.collect();
 	}
 
+	@Test
+	public void testStringConcat() throws Exception {
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		TableEnvironment tableEnv = new TableEnvironment();
+
+		DataSet<Tuple2<String, Integer>> ds = env.fromElements(
+			new Tuple2<>("ABCD", 3),
+			new Tuple2<>("ABCD", 2));
+
+		Table in = tableEnv.fromDataSet(ds, "a, b");
+
+		Table result = in
+			.select("a + b + 42");
```

Comment:
Maybe you need to rethink your approach. Expressions like `a + 42.abs().cast(STRING)` also do not work. It might be necessary to introduce a modified grammar where the dot is special symbol followed by symbols for `cast`, functions etc. without a dot.